frontend/src/components/mainPage/codeEditor/subcomponents/Codearea.vue

```diff
@@ -42,6 +43,72 @@ export default {
             this.$emit("pickVariableEvent", variable);
         },
 
+        moveTrackedVariables(changes) {
+            const handleVarTrackerMove = (ch, varObj) => {
+                if (ch.size > 0) {
+                    if (ch.start > varObj.start && ch.start < varObj.end) return "Delete";
+                    if (ch.start <= varObj.start) {
+                        varObj.start += ch.size;
+                        varObj.end += ch.size;
+                    }
+                } else {
+                    if (ch.start >= varObj.start && ch.start < varObj.end) return "Delete";
+                    if (ch.end > varObj.start && ch.end < varObj.end) return "Delete";
+                    if (ch.start <= varObj.start) {
+                        varObj.start += ch.size;
+                        varObj.end += ch.size;
+                    }
+                }
+            };
+
+            this.project.sceneObjects = this.project.sceneObjects.filter((sceneObj) => {
+                let mainVar = sceneObj.variable;
+                if (handleVarTrackerMove(changes, mainVar) == "Delete") {
+                    return false;
+                }
+
+                for (let subobjId in sceneObj.subobjects) {
+                    let subobj = sceneObj.subobjects[subobjId];
+                    let subVar = subobj.variable;
+                    if (handleVarTrackerMove(changes, subVar) == "Delete") {
+                        return false;
+                    }
+                }
+
+                return true;
+            });
+        },
+
+        moveBreakpoints(changes) {
+            if (changes.deltaLineCount == 0) return;
+
+            let firstChangedLine = (this.$props.code.substr(0, changes.start - 1).match(/\n/g) || []).length;
+
+            let affectedBreakpoints = [];
+            for (let bp of this.project.breakpoints) {
+                if (bp[0] < firstChangedLine) continue;
+                if (bp[0] > firstChangedLine && bp[0] < firstChangedLine - changes.deltaLineCount) {
+                    this.project.breakpoints.addOrDelete({
+                        id: bp[0],
+                    });
+                } else if (bp[0] > firstChangedLine) {
+                    affectedBreakpoints.push(bp[0]);
+                }
+            }
+            for (let affectedBreakpoint of affectedBreakpoints) {
+                this.project.breakpoints.addOrDelete({
+                    id: affectedBreakpoint,
+                });
+                this.project.breakpoints.addOrDelete({
+                    id: affectedBreakpoint + changes.deltaLineCount,
+                });
+            }
+        },
+
```

Comment:
Zmieniłbym nazwenictwo zmiennej changes na change, bo to jest w sumie pojedyncza zmiana (changes sugeruje też, że to jest tablica).